src/csharp/NativeMethods.cs

```diff
@@ -171,6 +177,25 @@ internal class NativeLib
                                                                               int /* int32_t */ token,
                                                                               out IntPtr /* const char** */ outStr);
 
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern IntPtr /* OgaResult* */ OgaCreateTensorFromBuffer(IntPtr /* data* */ data, long[] shape_dims, UIntPtr shape_dims_count, ElementType element_Type,
+                                                                               out IntPtr /* OgaTensor** */ tensor);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern void OgaDestroyTensor(IntPtr /* OgaTensor * */ tensor);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern IntPtr /* OgaResult* */ OgaTensorGetType(IntPtr /* OgaTensor * */ tensor, out ElementType element_type);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern IntPtr /* OgaResult* */ OgaTensorGetShapeSize(IntPtr /* OgaTensor * */ tensor, out UIntPtr size);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern IntPtr /* OgaResult* */ OgaTensorGetShape(IntPtr /* OgaTensor * */ tensor, long[] shape_dims, UIntPtr /* size_t */ shape_dims_count);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern IntPtr /* OgaResult* */ OgaTensorGetData(IntPtr /* OgaTensor * */ tensor, out IntPtr /* void* */ data);
```

Comment:
How does the caller know the length of the data without multiplying the shape?
Do you plan to expose it via Tensor class?

Reply:
We could certainly add it eventually. The current APIs were to allow people to pass data in, this could be added in the future as needed.